test/dataclasses/test_chat_message.py

```diff
@@ -0,0 +1,166 @@
+    assert not message.tool_call_result
+
+def test_from_system_with_valid_content():
+    text = "I have a question."
+    message = ChatMessage.from_system(text=text)
+
+    assert message.role == ChatRole.SYSTEM
+    assert message._content == [TextContent(text)]
+
+    assert message.text == text
+    assert message.texts == [text]
+
+    assert not message.tool_calls
+    assert not message.tool_call
+    assert not message.tool_call_results
+    assert not message.tool_call_result
+
+def test_from_tool_with_valid_content():
+    tool_result = "Tool result"
+    origin = ToolCall(id="123", tool_name="mytool", arguments={"a": 1})
+    message = ChatMessage.from_tool(tool_result, origin)
+
+    tcr = ToolCallResult(result=tool_result, origin=origin)
+
+    assert message._content == [tcr]
+    assert message.role == ChatRole.TOOL
+
+    assert message.tool_call_result == tcr
+    assert message.tool_call_results == [tcr]
+
+    assert not message.tool_calls
+    assert not message.tool_call
+    assert not message.texts
+    assert not message.text
+
+def test_multiple_text_segments():
+    texts = [TextContent(text="Hello"), TextContent(text="World")]
+    message = ChatMessage(_role=ChatRole.USER, _content=texts)
+
+    assert message.texts == ["Hello", "World"]
+    assert len(message) == 2
+
+def test_mixed_content():
+    content = [
+        TextContent(text="Hello"),
+        ToolCall(id="123", tool_name="mytool", arguments={"a": 1}),
+    ]
+
+    message = ChatMessage(_role=ChatRole.ASSISTANT, _content=content)
+
+    assert len(message) == 2
+    assert message.texts == ["Hello"]
+    assert message.text == "Hello"
+
+    assert message.tool_calls == [content[1]]
+    assert message.tool_call == content[1]
+
+def test_to_dict():
+    message = ChatMessage.from_user("content")
+    message.meta["some"] = "some"
+
+    assert message.to_dict() == {
+        "_content": [{"text": "content"}],
+        "_role": "user",
+        "_meta": {"some": "some"},
+    }
+
+
+def test_from_dict():
+    assert ChatMessage.from_dict(data={"_content": [{"text": "text"}], "_role": "user"}) == ChatMessage.from_user(text="text")
+
+
+def test_from_dict_with_meta():
+    data = {
+        "_content": [{"text": "text"}],
+        "_role": "assistant",
+        "_meta": {"something": "something"}
+    }
+
+    assert ChatMessage.from_dict(data=data) == ChatMessage.from_assistant(text="text", meta={"something": "something"})
+
+def test_serde_with_tool_calls():
+    tool_call = ToolCall(id="123", tool_name="tool", arguments={"a": 1})
+    message = ChatMessage.from_assistant("a message", tool_calls=[tool_call])
+
+    serialized_message = message.to_dict()
+
+    assert serialized_message == {
+        "_content": [{"text": "a message"}, {"tool_call": {"id": "123", "tool_name": "tool", "arguments": {"a": 1}}}],
+        "_role": "assistant",
+        "_meta": {},
+    }
+
+    assert ChatMessage.from_dict(serialized_message) == message
```

Comment:
Let's extend these tests to account for all content types, i.e., let's add each of the three supported payloads to the message and then perform to/from dict, etc. On the same note, we should also test for the exception handling (as suggested by the above comment).